conans/test/functional/toolchains/cmake/test_cmake_toolchain_m1.py

```diff
@@ -0,0 +1,65 @@
+import textwrap
+import platform
+
+
+import pytest
+
+from conans.test.assets.sources import gen_function_cpp
+from conans.test.utils.tools import TestClient
+
+
+@pytest.mark.skipif(platform.system() != "Darwin", reason="Only OSX")
+@pytest.mark.parametrize("op_system", ["Macos", "iOS"])
+def test_m1(op_system):
+    os_version = "os.version=12.0" if op_system == "iOS" else ""
+    profile = textwrap.dedent("""
+        include(default)
+        [settings]
+        os={}
+        {}
+        arch=armv8
+    """.format(op_system, os_version))
+
+    client = TestClient(path_with_spaces=False)
+    client.save({"m1": profile}, clean_first=True)
+    client.run("new hello/0.1 --template=v2_cmake")
+    client.run("create . --profile:build=default --profile:host=m1 -tf None")
+
+    main = gen_function_cpp(name="main", includes=["hello"], calls=["hello"])
+    cmakelists = textwrap.dedent("""
+    cmake_minimum_required(VERSION 3.15)
+    project(MyApp CXX)
+    set(hello_DIR "${CMAKE_BINARY_DIR}")
```

Comment:
Hopefully this ``hello_DIR`` should no longer be necessary?

Reply:
Yes, it is yet necessary to trick CMake (when cross-building on Apple) to search outside the system frameworks dirs.

Reply:
Ok, lets merge this, and address that in a separate issue.